src/ViewRenderer/ViewRenderer.php

```diff
@@ -108,37 +104,48 @@ public function withLayout(string $layout): self
         return $new;
     }
 
-    public function withCsrf(string $requestAttribute = Csrf::REQUEST_NAME): self
+    /**
+     * @param ContentParamsInjectionInterface[]|LayoutParamsInjectionInterface[]|LinkTagsInjectionInterface[]|MetaTagsInjectionInterface[] $injections
+     * @return self
+     */
+    public function addInjections(array $injections): self
     {
         $new = clone $this;
-        $new->csrfTokenRequestAttribute = $requestAttribute;
-        $new->csrfToken = $new->getCsrfToken();
+        $new->injections = array_merge($this->injections, $injections);
+        return $new;
+    }
 
+    /**
+     * @param ContentParamsInjectionInterface|LayoutParamsInjectionInterface|LinkTagsInjectionInterface|MetaTagsInjectionInterface $injection
+     * @return self
+     */
+    public function addInjection($injection): self
+    {
+        return $this->addInjections([$injection]);
+    }
+
+    /**
+     * @param ContentParamsInjectionInterface[]|LayoutParamsInjectionInterface[]|LinkTagsInjectionInterface[]|MetaTagsInjectionInterface[] $injections
+     * @return self
+     */
+    public function withInjections(array $injections): self
```

Comment:
What is the difference with addInjections?

Reply:
withInjection overrides injections, addInjection merge with exist

Reply:
withInjections replace injections to $injections.
addInjections add $injections to current injections (for exmaple in contact controller we make addInjection($this->csrfViewInjection) for add CSRF to injections).

Reply:
> withInjection overrides injections, addInjection merge with exist

Then, immutable methods should start with the `with` prefix, so I'd suggest rename addInjections to withAddedInjections.

Reply:
Done!